modules/nf-core/modules/diamond/blastx/main.nf

```diff
@@ -0,0 +1,53 @@
+process DIAMOND_BLASTX {
+    tag "$meta.id"
+    label 'process_medium'
+
+    // Dimaond is limited to v2.0.9 because there is not a
+    // singularity version higher than this at the current time.
+    conda (params.enable_conda ? "bioconda::diamond=2.0.9" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/diamond:2.0.9--hdcc8f71_0' :
+        'quay.io/biocontainers/diamond:2.0.9--hdcc8f71_0' }"
+
+    input:
+    tuple val(meta), path(fasta)
+    path db
+    val outext
+
+    output:
+    tuple val(meta), path('*.{blast,xml,txt,daa,sam,tsv,paf}'), emit: output
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    switch ( outext ) {
+        case "blast": outfmt = 0; break
+        case "xml": outfmt = 5; break
+        case "txt": outfmt = 6; break
+        case "daa": outfmt = 100; break
+        case "sam": outfmt = 101; break
+        case "tsv": outfmt = 102; break
+        case "paf": outfmt = 103; break
+    }
```

Comment:
I know this comes from the nf-core modules but isn't this something one would rather set with `ext.args` and have a default?

Reply:
Ideally yes, but the problem is that DIAMOND does not specify any file extension... _except_ for `daa`. 

So this makes it problematic when you want to specify _exactly_ what file comes out, and I would rather be explict which file is generated, so I use this information to save the file with an appropriate suffix. 

I could've specified the input channel for `--output-fmt` directly with the numeric codes, but agian for understandability purposes I thought it easier to use if you tell it exactly which one to make with a 'human readable' string.